addons/mail/models/discuss/discuss_channel.py

```diff
@@ -705,6 +706,23 @@ def _channel_message_notifications(self, message, message_format=False):
     #   - when a message is posted on a channel (to the channel, using _notify() method)
     # ------------------------------------------------------------
 
+    def toggle_allow_public_upload(self):
+        """ Toggle the allow_public_upload field of the channel and notify the
+        channel members"""
+        self.ensure_one()
+        self.allow_public_upload = not self.allow_public_upload
```

Comment:
prone to race conditions, I'd much prefer if each call explicitly passed the expected state
(if 2 person click on it to enable it, it would end up disabled for no reason)